java/location-line-of-sight/src/main/java/com/esri/arcgisruntime/sample/locationlineofsight/MainActivity.java

```diff
@@ -0,0 +1,115 @@
+import android.view.MotionEvent;
+
+import com.esri.arcgisruntime.geoanalysis.LineOfSight;
+import com.esri.arcgisruntime.geoanalysis.LocationLineOfSight;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.ArcGISTiledElevationSource;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.AnalysisOverlay;
+import com.esri.arcgisruntime.mapping.view.Camera;
+import com.esri.arcgisruntime.mapping.view.DefaultSceneViewOnTouchListener;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+
+public class MainActivity extends AppCompatActivity {
+
+  private SceneView mSceneView;
+  private AnalysisOverlay mAnalysisOverlay;
+  private Point mObserverLocation;
+  private Point mTargetLocation;
+
+  private final double mZOffset = 2.0;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // create a scene and add a basemap to it
+    ArcGISScene scene = new ArcGISScene();
+    scene.setBasemap(Basemap.createImagery());
+
+    // create SceneView from layout
+    mSceneView = findViewById(R.id.sceneView);
+    mSceneView.setScene(scene);
+
+    // create an elevation source, and add this to the base surface of the scene
+    ArcGISTiledElevationSource elevationSource = new ArcGISTiledElevationSource(getString(R.string.elevation_image_service));
+    scene.getBaseSurface().getElevationSources().add(elevationSource);
+
+    // create an analysis overlay to contain the analysis and add it to the scene view
+    mAnalysisOverlay = new AnalysisOverlay();
+    mSceneView.getAnalysisOverlays().add(mAnalysisOverlay);
+
+    // add a camera and initial camera position
+    Camera camera = new Camera(new Point(-73.10861935949697, -49.25758493899104, 3050), 106, 73, 0);
+    mSceneView.setViewpointCamera(camera);
+
+    // set the visible and obstructed colors (default would be green/red)
+    // these are static properties that apply to all line of sight analyses in the scene view
+    LineOfSight.setVisibleColor(Color.CYAN);
+    LineOfSight.setObstructedColor(Color.MAGENTA);
+
+    mSceneView.setOnTouchListener(new DefaultSceneViewOnTouchListener(mSceneView) {
+      @Override public boolean onSingleTapConfirmed(MotionEvent motionEvent) {
+
+        // get a screen point from the motion event
+        android.graphics.Point screenPoint = new android.graphics.Point(Math.round(motionEvent.getX()),
+            Math.round(motionEvent.getY()));
+
+        // convert the screen point to a scene point
+        Point scenePoint = mSceneView.screenToBaseSurface(screenPoint);
+
+        if (mObserverLocation == null) {
+          // set the observer location
+          mObserverLocation = new Point(scenePoint.getX(), scenePoint.getY(), scenePoint.getZ() + mZOffset, scenePoint.getSpatialReference());
```

Comment:
You can get an uncaught error here (and similar line below) as scenePoint can be null - e.g. if you tap off the horizon of the earth (e.g. there's no valid location to return).

Reply:
Good catch! Thanks @shellygill